bin/run_brick.rb

```diff
@@ -14,28 +14,65 @@ def get_brick_params(prefix)
   ENV.select { |k,| k.to_s.match(/^#{prefix}.*/) }.map { |k, v| [k.slice(prefix.length..-1), v] }.to_h
 end
 
+def log_execution_summary_error(params, log, brick_type, error_message)
+  result = {
+    errorCode: 'lcm.error',
+    message: error_message,
+    parameters: []
+  }
+
+  begin
+    log_directory = params['GDC_LOG_DIRECTORY']
+    execution_id = params['GDC_EXECUTION_ID']
+    execution_result_logger_file = params['GDC_EXECUTION_RESULT_LOG_PATH'].nil? ? "#{log_directory}/#{execution_id}_result.json" : params['GDC_EXECUTION_RESULT_LOG_PATH']
+    File.open(execution_result_logger_file, 'w') { |file| file.write(JSON.pretty_generate(result)) }
+    log.info "action=log_#{brick_type}_result execution_id=#{execution_id} status=success error_message=#{error_message}"
+  rescue Exception => e # rubocop:disable RescueException
+    log.info "action=log_#{brick_type}_result execution_id=#{execution_id} status=failed exception=#{e}"
+  end
+end
+
+def handle_error(params, log, brick_type, error, error_message)
+  execution_log = GoodData.logger
+  execution_log.error "Execution failed. Error: #{error}" unless execution_log.nil?
+  log_execution_summary_error(params, log, brick_type, error_message)
+  log.info "action=#{brick_type}_execution status=failed commit_hash=#{params['GOODDATA_RUBY_COMMIT']} execution_id=#{params['GDC_EXECUTION_ID']} exception=#{error}"
+  raise
+end
+
 syslog_node = ENV['NODE_NAME']
 log = RemoteSyslogLogger.new(syslog_node, 514, :program => "ruby_#{brick_type}", :facility => 'local2')
 
 log.info "action=#{brick_type}_execution status=init"
 
 begin
+  commit_hash = ENV['GOODDATA_RUBY_COMMIT'] || ''
+  execution_id = ENV['GDC_EXECUTION_ID']
+  log_directory = ENV['GDC_LOG_DIRECTORY'] || '/tmp/'
+  error_log_params = {
+    GOODDATA_RUBY_COMMIT: commit_hash,
+    GDC_EXECUTION_ID: execution_id,
+    GDC_LOG_DIRECTORY: log_directory,
+    GDC_EXECUTION_LOG_PATH: params['GDC_EXECUTION_LOG_PATH'],
+    GDC_EXECUTION_RESULT_LOG_PATH: params['GDC_EXECUTION_RESULT_LOG_PATH']
+  }
+
   brick_pipeline = GoodData::Bricks::Pipeline.send("#{brick_type}_pipeline")
   normal_params = get_brick_params(BRICK_PARAM_PREFIX)
   hidden_params = get_brick_params(HIDDEN_BRICK_PARAMS_PREFIX)
   params = normal_params.merge(hidden_params)
 
   params['values_to_mask'] = hidden_params.values
-  commit_hash = ENV['GOODDATA_RUBY_COMMIT'] || ''
-  execution_id = ENV['GDC_EXECUTION_ID']
   params['gooddata_ruby_commit'] = commit_hash
-  params['GDC_LOG_DIRECTORY'] = ENV['GDC_LOG_DIRECTORY'] || '/tmp/'
+  params['GDC_LOG_DIRECTORY'] = log_directory
   params['GDC_EXECUTION_ID'] = execution_id
+  params['GDC_EXECUTION_LOG_PATH'] = ENV['GDC_EXECUTION_LOG_PATH']
+  params['GDC_EXECUTION_RESULT_LOG_PATH'] = ENV['GDC_EXECUTION_RESULT_LOG_PATH']
+
   log.info "action=#{brick_type}_execution status=start commit_hash=#{commit_hash} execution_id=#{execution_id}"
   brick_pipeline.call(params)
```

Comment:
We should update execution result for OK case too, update_execution_result(OK, "")